### PR TITLE
Fix: Fixed issue where the drive letter was included in the name

### PR DESCRIPTION
**Pull request by cinqmilleans**
**Resolved / Related Issues**
Improves PropertiesGeneral control:
- The code is more readable. Some warnings are fixed.
- When leaving the name field with an empty name, the previous name is returned.
- For drives, the letter of the name is removed during editing. #10263.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**

https://user-images.githubusercontent.com/46631671/198020958-e7927c35-3a52-45ce-a73c-1c6c67da5216.mp4

### CODE REVIEW

File: src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs [hunk 1]
```diff
@@ -1,113 +1,131 @@
...
+				var drive = driveProps.Drive;
+				newName = letterRegex.Replace(newName, string.Empty); // Remove "(C:)" from the new label
+
+				Win32API.SetVolumeLabel(drive.Path, newName);
+				_ = App.Window.DispatcherQueue.EnqueueAsync(async () =>
+				{
+					await drive.UpdateLabelAsync();
+					await fsVM.SetWorkingDirectoryAsync(drive.Path);
+				});
+				return true;
```
Comment (QuaintMako) [inline src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs:48]: Maybe extract the `if` instructions into a method to make `SaveChangesAsync ` smaller and easier to read?

File: src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs [hunk 2]
```diff
@@ -1,113 +1,131 @@
...
+		private void ItemFileName_LosingFocus(UIElement _, LosingFocusEventArgs e)
+		{
+			if (string.IsNullOrWhiteSpace(ItemFileName.Text))
+				ItemFileName.Text = ViewModel.ItemName;
+			else
+			{
+				var match = letterRegex.Match(ViewModel.OriginalItemName);
+				if (match.Success)
+					ItemFileName.Text += match.Value;
+			}
```
Comment (QuaintMako) [inline src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs:122]: The `else` could be eliminated if we add a `return` inside the first `if`.
```suggestion
			if (string.IsNullOrWhiteSpace(ItemFileName.Text)) 
			{
				ItemFileName.Text = ViewModel.ItemName;
				return;
			}

			var match = letterRegex.Match(ViewModel.OriginalItemName);
			if (match.Success)
				ItemFileName.Text += match.Value;
```

**Review by QuaintMako — DISMISSED** [review]
LGTM, just a few ideas to make the code a tad cleaner.
Could be integrated as it is.

**Comment by cinqmilleans** [comment]
I refactored the method with sub-methods.

**Review by yaichenbaum — APPROVED** [review]

**Pull request merged**